test/integration/containerized/test_cleanup_images.py

```diff
@@ -0,0 +1,58 @@
+import shutil
+import json
+from base64 import b64decode
+import os
+
+import pytest
+
+from ansible_runner.cleanup import cleanup_images, prune_images
+from ansible_runner.defaults import default_container_image
+
+
+@pytest.mark.parametrize('runtime', ['podman', 'docker'])
+def test_cleanup_new_image(cli, runtime, tmp_path):
+    if shutil.which(runtime) is None:
+        pytest.skip(f'{runtime} is unavaialble')
+
+    # Create new image just for this test with a unique layer
+    special_string = "Verify this in test - 1QT4r18a7E"
+    dockerfile_path = str(tmp_path / 'Dockerfile')
+    with open(dockerfile_path, 'w') as f:
+        f.write('\n'.join([
+            'FROM {}'.format(default_container_image),
+            'RUN echo {} > /tmp/for_test.txt'.format(special_string)
+        ]))
+    image_name = 'quay.io/fortest/hasfile:latest'
```

Comment:
If this were a fixture used by multiple tests I would have randomized the name. But why would the _same_ test run in parallel on the _same_ host? Yes, tests will run in parallel but I imagined it would never run the same test more than once. I still don't have a problem adding randomization because it won't hurt anything, I just wanted to explain myself.

Reply:
The other thing is multiple instances of the same tests with different Python versions running at the same time.